### PR TITLE
Issue #89 - Environment setup script

### LINKED ISSUE #89: EPIC: Appropriately handling the inclusion of default configuration files in source control
While writing the section [Automating memory checking on MacOS using `leaks`](https://github.com/graph-algorithms/edge-addition-planarity-suite/wiki/3.-Test-Support#automating-memory-checking-on-macos-using-leaks) of the [3. Test Support](https://github.com/graph-algorithms/edge-addition-planarity-suite/wiki/3.-Test-Support) wiki page, the issue of how to handle the inclusion of default files in source control arose: in #59 , I added the `TestSupport/planaritytesting/leaksorchestrator/planarity_leaks_config.ini` file (generated by the `planarity_leaks_config_manager.py`) to source control, and then added a corresponding line to `.gitignore`, ([erroneously](https://git-scm.com/docs/gitignore#_notes)) assuming that even if a file was tracked in source control, the `.gitignore` would ensure that no changes could be made to the file without first changing the `.gitignore`.

After observing that the behaviour didn't match my expectation, I did some more investigation and arrived at an intermediate solution: instructing users to locally run [`git update-index --assume-unchanged`](https://git-scm.com/docs/git-update-index#Documentation/git-update-index.txt---no-assume-unchanged) on the defaults tracked in the repository so that local changes wouldn't appear as candidates to stage unless they actually intended to update the default in the repository (see also [`git update-index` - Using "Assume Unchanged" Bit](https://git-scm.com/docs/git-update-index#_using_assume_unchanged_bit) for justification vs. using `--skip-worktree`)

However, this is not scalable when dealing with multiple default configuration files, and an entirely process-driven approach is prone to user error. [This StackOverflow response](https://stackoverflow.com/a/18571577/5456879) helps to inform a solution:

1. Create a directory containing default configuration files to be tracked in source control,
2. Implement an environment setup script to copy the default configuration files to the correct locations in the user environment:
    * `.vscode/` - `tasks.json`, `launch.json`, and `settings.json`
    * `TestSupport/planaritytesting/` - `.pylintrc` and `leaksorchestrator/planarity_leaks_config.ini`
3. Run `git -rm --cached` for the configuration files currently tracked in version control (especially the `.cproject`, `c/.cproject` and `.project` files)
4. Add those paths in the user development environment to the `.gitignore`
5. Include instructions in `README.md` to run the environment setup script

### PR DESCRIPTION
**Pull request by wbkboyer**
Resolves #89 

## Added
* `devEnvSetupAndDefaults/`
  * `devEnvSetup.sh` - copies default configuration files to expected target locations
  * `.vscode/` - contains default configuration files for `tasks.json`, `launch.json`, and `settings.json` (see #16 , #58 , and #92 )
  * `TestSupport/planaritytesting/`
    * `leaksorchestrator/planarity_leaks_config.ini` - Default configuration file for `TestSupport/planaritytesting/leaksorchestrator/planarity_leaks_orchestrator.py` script, generated by `planarity_leaks_config_manager.py` (see #59 and #81)
    * `.pylintrc` - configures (#100 will add more complete documentation to the [2. Dev Setup](https://github.com/graph-algorithms/edge-addition-planarity-suite/wiki/2.-Dev-Setup) wiki page)

## Updated
* `.gitignore` - ignore `.vscode` directory when it appears at *root* of repository; ignore Eclipse configuration files (if present); ignore `.pylintrc` and `planarity_leaks_config.ini` under `TestSupport/` directory
* `README.md` - Extracted development environment setup instructions and now refer to [2. Dev Setup](https://github.com/graph-algorithms/edge-addition-planarity-suite/wiki/2.-Dev-Setup) wiki page; minor tweaks and clarifications

## Removed
* `.cproject`, `.project`, and `c/.cproject` Eclipse IDE configuration files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,99 +18,56 @@ These instructions will get you a copy of the project up and running on your loc
 
 ### Pre-compiled Executable Releases for Non-Developers
 
-On several distributions of Linux, you may be able to get the planarity executable with _sudo apt-get install planarity_ , or you may already have it you have Matlab. For non-developer Windows users, there is also a pre-compiled executable version of the algorithm implementations. Download and decompress the _planarity-N.N.N.N.WindowsExe.zip_ file. 
+On several (Debian-based) distributions of Linux, you may be able to get the planarity executable with `sudo apt install planarity`, or you may already have it you have Matlab. For non-developer Windows users, there is also a pre-compiled executable version of the algorithm implementations: download and decompress the `planarity-N.N.N.N.WindowsExe.zip` file.
```
Comment (john-boyer-phd) [inline README.md:21]: ... you may already have it _if_ you have Matlab...

File: README.md [hunk 2]
```diff
@@ -18,99 +18,56 @@ These instructions will get you a copy of the project up and running on your loc
...
 
-The _planarity_ executable program also supports an extensive list of command-line parameters that make it possible to automate the execution of any of the algorithms included in the application. Run _planarity_ with the "-h" command-line parameter to get more information about the command line options, and use "-h -menu" for more extensive information about command-line mode. Essentially, all functionality available in the interactive, menu-driven mode is also available via the command-line parameters.
+The `planarity` executable program also supports an extensive list of command-line parameters that make it possible to automate the execution of any of the algorithms included in the application. Run `planarity` with the `-h` command-line parameter to get more information about the command line options, and use `-h -menu` for more extensive information about command-line mode. Essentially, all functionality available in the interactive, menu-driven mode is also available via the command-line parameters.
```
Comment (john-boyer-phd) [inline README.md:25]: Thanks for fixing all of the underscored items to put tick marks around them.

File: README.md [hunk 3]
```diff
@@ -18,99 +18,56 @@ These instructions will get you a copy of the project up and running on your loc
...
-
-Now that the project is available, the code can be built and executed:
-
-1. Open the C/C++ Perspective
-     1. Use the Open Perspectives button (or use Windows | Perspective | Open Perspective | Other...)
-     2. Select C/C++
-2. Right-click Planarity-C project, Build Configurations, Build All
-3. Right-click Planarity-C project, Build Configurations, Set Active, Release
-4. Right-click Planarity-C project, Run As, Local Application, planarity.exe (release)
+Please refer to the [2. Dev Setup](https://github.com/graph-algorithms/edge-addition-planarity-suite/wiki/2.-Dev-Setup) wiki page for instructions on how to install development dependencies on Windows 10 22H2, Debian 12.70, or MacOS Sequoia 15.0, as
```
Comment (john-boyer-phd) [inline README.md:29]: The readme should probably leave the versions of OS's to the wiki page. Also, on the wiki page, I think it's going to be reasonable to include Windows 11 not just Windows 10 22H2 (but that is a discussion about the wiki page that can happen outside of PR land).

File: README.md [hunk 4]
```diff
@@ -18,99 +18,56 @@ These instructions will get you a copy of the project up and running on your loc
...
 
-The result is a validated _planarity-N.N.N.N.tar.gz_ distribution, where _N.N.N.N_ is the version number expressed in the _configure.ac_ file. 
+The result is a validated `planarity-N.N.N.N.tar.gz` distribution, where `N.N.N.N` is the version number expressed in the `configure.ac` file. 
 
 ### Making and Running the Software from the Distribution
 
 If you have done the steps to set up the development environment and work with the code, then you can make and run the software using the development environment, so you don't necessarily need to make or run the software using the process below.
 
-You also don't necessarily need to make and install the planarity software on Linux if you are able to get it using _sudo apt-get planarity_ . 
+You also don't necessarily need to `make` and `make install` the planarity software on Linux if you are able to get it using `sudo apt planarity` (i.e. are using a Debian-based Linux distribution which uses [`apt`](https://en.wikipedia.org/wiki/APT_(software)) for package management)
```
Comment (john-boyer-phd) [inline README.md:50]: ... (i.e., using... [add comma, remove "are"]

**Review by john-boyer-phd — CHANGES_REQUESTED** [review]
Aside from a few minor tweaks given inline, let's discuss the following in stand-up tomorrow:
1. The PR shows the removal of the eclipse project files (good) but it doesn't seem to show me the removal of the VS code json files that are now templatized in the `devEnvSetupAndDefaults` directory. What will cause the non-templatized ones to be removed from master when this PR is merged?
2. In the `devEnvSetupAndDefaults`, there is now a `settings.json`, but there doesn't seem to be one in the old non-templatized `.vscode` directory. It's undoubtedly a good idea to be adding one, but I'd just like to talk through what it is contributing that we didn't have  across dev environments before.

**Comment by wbkboyer** [comment]
> Aside from a few minor tweaks given inline, let's discuss the following in stand-up tomorrow:
> 
> 1. The PR shows the removal of the eclipse project files (good) but it doesn't seem to show me the removal of the VS code json files that are now templatized in the `devEnvSetupAndDefaults` directory. What will cause the non-templatized ones to be removed from master when this PR is merged?

I did run `git rm --cached -r .vscode/`, but I think since I'm including the files in the `devEnvSetupAndDefaults/` directory, it's counting those files as "renamed" 🚀 

> 2. In the `devEnvSetupAndDefaults`, there is now a `settings.json`, but there doesn't seem to be one in the old non-templatized `.vscode` directory. It's undoubtedly a good idea to be adding one, but I'd just like to talk through what it is contributing that we didn't have  across dev environments before.

The workings of `settings.json` will be documented in [2. Dev Setup](https://github.com/graph-algorithms/edge-addition-planarity-suite/wiki/2.-Dev-Setup) as part of #100 🎉

**Review by john-boyer-phd — APPROVED** [review]
Looks great, thanks!

**Pull request merged**